### PR TITLE
Ported ExampleCustomCurrency.cs into 1.4.

### PR DESCRIPTION
**Pull request by Vaema**
I've made ExampleCustomCurrency for 1.4 Example Mod.

### CODE REVIEW

File: ExampleMod/Content/Currencies/ExampleCustomCurrency.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+using Microsoft.Xna.Framework;
+using Terraria;
+using Terraria.GameContent.UI;
+using Terraria.Localization;
+
+namespace ExampleMod
+{
+	public class ExampleCustomCurrency : CustomCurrencySingleCoin
+	{
+		public static readonly Color ExampleCustomCurrencyTextColor = Color.BlueViolet; // This variable is located in the GetPriceText hook.
```
Comment (RighteousRyan1) [inline ExampleMod/Content/Currencies/ExampleCustomCurrency.cs:10]: Why is this necessary?

**Comment by Steviegt6** [comment]
Have we even verified custom currencies work properly?

**Comment by JavidPack** [comment]
I seem to have messed something up, I'll handle this.

**Comment by JavidPack** [comment]
Ok, I'm just going to do all the rest of the code on the 1.4 branch directly and mention you as a coauthor. Git seemed to mess up the commit history so I wasn't able to rebase to 1.4 successfully. Thanks for this.
